tests/flow/test_ts_delete.py

```diff
@@ -120,3 +120,99 @@ def test_bad_del(self):
         self.assertTrue(r.execute_command("SET", "BAD_X", "NOT_TS"))
         with pytest.raises(redis.ResponseError) as excinfo:
             r.execute_command("TS.DEL", "BAD_X", 100, 200)
+
+def test_del_retention_with_rules(self):
+    sample_len = 1010
+    with Env().getClusterConnectionIfNeeded() as r:
+        r.execute_command("ts.create", 'test_key_2', 'RETENTION', 1, 'uncompressed')
+        r.execute_command("ts.create", 'test_key_3', 'uncompressed')
+        r.execute_command('ts.createrule', 'test_key_2', 'test_key_3', 'AGGREGATION', 'avg', 10)
+
+        for i in range(sample_len):
+            assert i == r.execute_command("ts.add", 'test_key_2', i, 1)
+
+        with pytest.raises(redis.ResponseError) as excinfo:
+            r.execute_command("ts.del", "test_key_2", 1, 10)
+
+        r.execute_command("ts.create", 'test_key_4', 'RETENTION', 25, 'uncompressed')
+        r.execute_command("ts.create", 'test_key_5', 'uncompressed')
+        r.execute_command('ts.createrule', 'test_key_4', 'test_key_5', 'AGGREGATION', 'avg', 10)
+
+        for i in range(30):
+            assert i == r.execute_command("ts.add", 'test_key_4', i, 1)
+
+        with pytest.raises(redis.ResponseError) as excinfo:
+            r.execute_command("ts.del", "test_key_4", 9, 10)
+
+
+def test_del_with_rules(self):
+    sample_len = 1010
+    e = Env()
+    with e.getClusterConnectionIfNeeded() as r:
+        r.execute_command("ts.create", 'test_key_2', 'RETENTION', 5000, 'uncompressed')
+        r.execute_command("ts.create", 'test_key_3', 'uncompressed')
+        r.execute_command('ts.createrule', 'test_key_2', 'test_key_3', 'AGGREGATION', 'sum', 10)
+
+        for i in range(70):
+            assert i == r.execute_command("ts.add", 'test_key_2', i, 1)
+        for i in range(80, sample_len):
+            assert i == r.execute_command("ts.add", 'test_key_2', i, 1)
+
+        res = r.execute_command('ts.range', 'test_key_3', 0, 9)
+        e.assertEqual(len(res), 1)
+        assert res[0] == [0, b'10']
+        assert r.execute_command("ts.del", "test_key_2", 0, 9) == 10
```

Comment:
love the fact you assert on this, it makes it easier to understand the code plus verify we are testing what we think we are testing.

Reply:
TX!